sdk/proto/provider.proto

```diff
@@ -177,6 +177,7 @@ message DiffResponse {
 message CreateRequest {
     string urn = 1;                        // the Pulumi URN for this resource.
     google.protobuf.Struct properties = 2; // the provider inputs to set during creation.
+    string timeout = 3;                    // the create request timeout represented as a string e.g. 5m.
```

Comment:
Feels like these should be numbers of seconds or milliseconds, and that it should be the responsibility of the engine to parse the string values that come in with the RegisterResourceRequest.

Reply:
(putting the parsing in the engine rather than in each provider ought to help ensure that all time values use a uniform format)